### ISSUE TITLE
type resolution fails in walkSelectorExpr

### ISSUE THREAD

**Issue by DQNEO** [DQNEO/babygo#91]
# Branch

https://github.com/DQNEO/babygo/tree/dev

# How to reproduce the bug
```
$  ./docker-run 
# make test
```

# Error message
```
# t/test.go:332: walkSelectorExpr String
#   looking for struct field or method "String"
#   X type is *ast.Ident
panic: Unexpected struct type *ast.Ident

goroutine 1 [running]:
main.getTypeOfExprAst({0x4b5b60?, 0xc00024b920?})
	/root/go/src/github.com/DQNEO/babygo/main.go:2754 +0x13a5
main.walkSelectorExpr(0xc00024b920, 0x0?)
	/root/go/src/github.com/DQNEO/babygo/main.go:4044 +0x49c
main.walkExpr({0x4b5b60?, 0xc00024b920?}, 0xc0001c0b90?)
	/root/go/src/github.com/DQNEO/babygo/main.go:4606 +0x2d4
main.walkCallExpr(0xc000413d70, 0x0?)
	/root/go/src/github.com/DQNEO/babygo/main.go:4072 +0x113
main.walkExpr({0x4b54e0?, 0xc000413d70?}, 0x40b28a?)
	/root/go/src/github.com/DQNEO/babygo/main.go:4608 +0x314
main.walkCompositeLit(0xc00034c660, 0x1?)
	/root/go/src/github.com/DQNEO/babygo/main.go:4340 +0x808
main.walkExpr({0x4b5560?, 0xc00034c660?}, 0xc00030d240?)
	/root/go/src/github.com/DQNEO/babygo/main.go:4602 +0xa9
main.prepareArgs(0x4bc780?, {0x0?, 0x0}, {0xc00030fda0?, 0x2, 0xc00030d240?}, 0x0)
	/root/go/src/github.com/DQNEO/babygo/main.go:615 +0x993
main.walkCallExpr(0xc000413e00, 0xc0001c1390?)
	/root/go/src/github.com/DQNEO/babygo/main.go:4239 +0x1865
main.walkExpr({0x4b54e0?, 0xc000413e00?}, 0x575660?)
	/root/go/src/github.com/DQNEO/babygo/main.go:4608 +0x314
main.walkExprStmt(...)
	/root/go/src/github.com/DQNEO/babygo/main.go:3336
main.walkStmt({0x4b56a0?, 0xc000495be0?})
	/root/go/src/github.com/DQNEO/babygo/main.go:3916 +0x22e
main.walk(0xc0000ac0b0)
	/root/go/src/github.com/DQNEO/babygo/main.go:4889 +0x1b87
main.compile(0xc0001c1d68, 0xc00018e1e0, {0x4ce617, 0x4}, {0x4ce617, 0x4}, {0xc00042b940, 0x2, 0x413c0d?}, {0x0, ...}, ...)
	/root/go/src/github.com/DQNEO/babygo/main.go:5345 +0x67e
main.buildAll({0xc0000140a0, 0x2, 0x0?})
	/root/go/src/github.com/DQNEO/babygo/main.go:5455 +0x8fa
main.main()
	/root/go/src/github.com/DQNEO/babygo/main.go:5389 +0x189
make: *** [Makefile:27: /tmpfs/bbg/bbg-test.d] Error 2
```

# the source the compiler fails to compile

in `t/test.go` ,
```
func testTokenString() {
	tok := token.Token("hello")
	fmt.Printf("%s\n", tok.String())
}
```

It seems like the compiler fails to find the return type of `tok.String()`

# the compiler's problem

maybe the logic around here has the issue

`main.go`

```
			switch typ := ut.E.(type) {
			case *ast.StructType: // strct.field
				structTypeLiteral = typ
			case *ast.StarExpr: // ptr.field
				structType := e2t(typ.X)
				structTypeLiteral = getUnderlyingStructType(structType)
				logf("  struct type is %s\n", serializeType(structType))
			default:
				panic(fmt.Sprintf("Unexpected struct type %T", ut.E))
			}
```

It fails to catch `*ast.Ident` ? I'm not sure right now...

**Comment by DQNEO** [comment]
@SandboxHead Thanks for the suggestion. For now I don't mean to assign this to anyone.  Just you can make a pull request regardless of assignments.

**Comment by DQNEO** [comment]
Fixed by 1d86dbc443072e8c161222a06d0bae9065eb9c43
